app/views/admin/courses/_form.html.erb

```diff
@@ -1,18 +1,20 @@
-<%= form_with model: [:admin, @course],
+<%= form_with model: @course.nil? ? @current_course : [:admin, @course],
```

Comment:
Now that this form is shared across two different methods, please do the following:

1. Modify the admin controller so that it uses `@current_course` instead of `@course`.
2. Move this file into the more standard `views/courses/` folder, instead of `admin`. I think we'll keep the shared components within the conventional folders rather than the `admin` folder.